## Problem 1: Mobile Recharge Validation System

Create function to validate mobile recharge request

In [1]:
def mobile_recharge():
    valid_plans = [199, 299, 399, 599]

    # Runs a while loop till correct amount is selected
    while True:
        amount = int(input("Enter a recharge amount: "))
        if amount < 50:
            print("Try again")
        elif amount not in valid_plans:
            print("Please select a valid plan of 199, 299, 399 or 599")
            print("Try again")

        # Valid rechrage
        else:
            print("Recharge Successful")
            break
            
mobile_recharge()

Enter a recharge amount:  40


Try again


Enter a recharge amount:  60


Please select a valid plan of 199, 299, 399 or 599
Try again


Enter a recharge amount:  199


Recharge Successful


#### Used test input:
'''
Enter a recharge amount:  40

Try again

Enter a recharge amount:  60

Please select a valid plan of 199, 299, 399 or 599

Try again

Enter a recharge amount:  199

Recharge Successful
'''

## Problem 2: Inventory Reorder Alert System

Create function to check for stock and reorder if necessary

In [2]:
# dictionary of items and their quantities
warehouse = {
    "Books":100,
    "Mobiles":5,
    "Electronic Accessories":70,
    "Medical devices":14,
    "Paper Supplies":70,
    "Games":15
}


def check_stock():
    print("--------------------------------------")
    print("   Status   |   Item   |   Quantity   ")
    print("--------------------------------------")

    # Checks item quantities
    for stock in warehouse:
        if warehouse[stock] < 15:
            print(f"Reorder Alert → {stock} : {warehouse[stock]}")
        else:
            print(f"Stock OK      → {stock} : {warehouse[stock]}")

check_stock()

--------------------------------------
   Status   |   Item   |   Quantity   
--------------------------------------
Stock OK      → Books : 100
Reorder Alert → Mobiles : 5
Stock OK      → Electronic Accessories : 70
Reorder Alert → Medical devices : 14
Stock OK      → Paper Supplies : 70
Stock OK      → Games : 15


## Problem 3: Student Result Processing System

Create funtion to process Pass/Fail for students based on average marks

In [3]:
# Simple function with list already given
def student_results(marks):
    total = 0
    for i in marks:
        total += i
    average = total/len(marks)
    if average >= 50:
        return "Pass"
    else:
        return "Fail"

# main
student = [17, 99, 57, 78, 63]
res = student_results(student)
print(res)

Pass


In [4]:
# Function with user input
def student_results(marks):
    total = 0
    for i in marks:
        total += i
    average = total/len(marks)
    if average >= 50:
        return "Pass"
    else:
        return "Fail"

# main
user_input = list(map(int, input("Enter marks of student: ").split()))
res = student_results(user_input)
print(f"The student has '{res}ed'")

Enter marks of student:  17 99 57 78 63


The student has 'Passed'


#### Used test input:
'''
Enter marks of student:  17 99 57 78 63

The student has 'Passed'
'''

## Problem 4: Cab Fare Estimator with Retry Option

Create function with fare calculator and retry option

In [5]:
def fare_estimator(total_km, hour_type):
    base_fare = 50
    cost_per_km = 12

    # Peak hour cost calculation
    cost = base_fare + (cost_per_km * total_km)

    # Check for 'peak hour'
    if hour_type == "peak hour":
        cost = cost * 1.25
        
    return cost

# main
while True:
    print("-----Fare Calculator-----")
    print("1.Calculate fare")
    print("2. Exit")
    user_input = int(input("Enter a choice: "))
    if user_input == 1:
                    
        total_km = float(input("Enter total km travelled: "))
        hour_type = input("Enter 'regular hour' or 'peak hour': ")

        # check for positive km distance
        if total_km < 1:
            print("Enter positive km!\n")
            continue

        # Get total fare
        res = fare_estimator(total_km, hour_type)
        print(f"\nYour fare is: {res}\n")

    # Exit condition
    if user_input == 2:
        print("Exiting successfully")
        break

-----Fare Calculator-----
1.Calculate fare
2. Exit


Enter a choice:  1
Enter total km travelled:  -1
Enter 'regular hour' or 'peak hour':  peak hour


Enter positive km!

-----Fare Calculator-----
1.Calculate fare
2. Exit


Enter a choice:  1
Enter total km travelled:  20
Enter 'regular hour' or 'peak hour':  regular hour



Your fare is: 290.0

-----Fare Calculator-----
1.Calculate fare
2. Exit


Enter a choice:  1
Enter total km travelled:  20
Enter 'regular hour' or 'peak hour':  peak hour



Your fare is: 362.5

-----Fare Calculator-----
1.Calculate fare
2. Exit


Enter a choice:  2


Exiting successfully


#### Used test input:
'''
-----Fare Calculator-----
#1.Calculate fare
2. Exit
Enter a choice:  1

Enter total km travelled:  -1

Enter 'regular hour' or 'peak hour':  peak hour

Enter positive km!

-----Fare Calculator-----
1.Calculate fare
2. Exit
Enter a choice:  1

Enter total km travelled:  20

Enter 'regular hour' or 'peak hour':  regular hour

Your fare is: 290.0

-----Fare Calculator-----
1.Calculate fare
2. Exit
Enter a choice:  1

Enter total km travelled:  20

Enter 'regular hour' or 'peak hour':  peak hour

Your fare is: 362.5

-----Fare Calculator-----
1.Calculate fare
2. Exit
Enter a choice:  2

Exiting successfully
'''

## Problem 5: Employee Attendance Eligibility Checker

Create function to check employee eligibilty based on attendance

In [6]:
def eligibility_checker(attendance):

    # calculate total present days
    present_days = 0
    for i in attendance:
        if i == "P":
            present_days += 1

    # calculate average percent
    present_percentage = (present_days/len(attendance)) * 100

    # Condition for eligibility
    if present_percentage >= 75:
        return "Eligible"
    else:
        return "Not Eligible"

# main
attendance = list(input("Enter 'P' for Present and 'A' for Absent: ").split())
res = eligibility_checker(attendance)
print(res)

Enter 'P' for Present and 'A' for Absent:  P P P A P


Eligible


#### Used test input:
'''
Enter 'P' for Present and 'A' for Absent:  P P P A P

Eligible
'''

## Problem 6: Password Strength Checker

Crete function to check password strength based on rules

In [7]:
def password_strength(password):
    # check for character length
    if len(password) < 8:
        return "Password must be atleast 8 characters"
        
    else:
        digit_present = False
        special = False

        # check digits and special characters
        for i in password:
            if i.isdigit():
                digit_present = True
            if i == "@" or i == "#" or i == "$":
                special = True

            # return strong password
            if digit_present and special:
                return "Passowrd is strong"
                
        # Conditions for when one or more rule fails
        if not digit_present:
            print("Password must contain atleast one digit")
        if not special:
            print("Password must contain atleast one special character '@#$'")
        return "Password is weak"

# main
password = input("Enter password: ")
res = password_strength(password)
print(res)

Enter password:  qwe@123r


Passowrd is strong


#### Used test input:
'''
Enter password:  qwe@123r

Passowrd is strong
'''